src/native/mqtt_connection.c

```diff
@@ -554,7 +588,8 @@ static void s_deliver_ack_failure(struct mqtt_jni_async_callback *callback, int
     AWS_FATAL_ASSERT(callback->connection);
 
     if (callback->async_callback) {
-        JNIEnv *env = aws_jni_get_thread_env(callback->connection->jvm);
+        /********** JNI ENV ACQUIRE **********/
+        JNIEnv *env = aws_jni_acquire_thread_env(callback->connection->jvm);
```

Comment:
`s_deliver_ack_failure()` is a helper function, not a callback, we could just pass env in as an argument